src/commands/init/init.py

```diff
@@ -1,53 +1,128 @@
+        )
 
-    if lib_dir and not lib_pth.is_dir():
-        lib_pth.mkdir(parents=True)
+    def create(self):
+        self.interactive_input()
+        self.project_creation()
 
-    project.write_config(
-        ProjectConfig(
+    def interactive_input(self):
+        project_name = ProjectCreator.request_input("Project name")
+        while project_name == "":
+            project_name = ProjectCreator.request_input_warning(
+                "Please provide a non-empty project name"
+            )
+        project_description = ProjectCreator.request_input("Project description")
+        author = ProjectCreator.request_input("Author")
+        version = ProjectCreator.request_input("Version")
+        project_license = ProjectCreator.request_input("License")
+        lib_dir = (
+            ProjectCreator.request_input("Libraries directory name (optional)") or "lib"
+        )
+        self.config = ProjectConfig(
             name=project_name,
             description=project_description,
             license=project_license,
             version=version,
             authors=[author],
             libs_path=lib_dir,
         )
-    )
 
-    Repo.init(project_root)
+    def project_creation(self):
+        project_root = Path() / self.config.name
+        self.copy_template(self.script_root, "default", project_root)
+        project = Project(project_root=project_root)
+
+        lib_pth = Path(project_root, self.config.libs_path)
+
+        if self.config.libs_path and not lib_pth.is_dir():
+            lib_pth.mkdir(parents=True)
+
+        project.write_config(self.config)
+
+        Repo.init(project_root)
 
+    @staticmethod
+    def copy_template(script_root: str, template_name: str, project_path: Path):
+        template_path = f"{script_root}/templates/{template_name}"
+        shutil.copytree(template_path, project_path)
+
+
+class OnlyConfigCreator(ProjectCreator):
+    def project_creation(self):
+        project_root = Path()
+
+        lib_pth = Path(project_root, self.config.libs_path)
+
+        if self.config.libs_path and not lib_pth.is_dir():
+            lib_pth.mkdir(parents=True)
+
+        project = Project(project_root=project_root)
+        project.write_config(self.config)
+
+        try:
+            Repo(project_root)
+        except InvalidGitRepositoryError:
+            Repo.init(project_root)
+
+
+def get_creator(args: Any) -> Type[ProjectCreator]:
+    if args.existing:
+        return OnlyConfigCreator
+
+    files_depth_3 = glob.glob("*") + glob.glob("*/*") + glob.glob("*/*/*")
+    can_be_a_project = any(
+        map(lambda f: f.endswith(".cairo") or f == ".git", files_depth_3)
+    )
+
+    out = False
+    if can_be_a_project:
+        out = input_yes_no(
+            "Your current directory may be a cairo project.\n"
+            "Do you want to adapt current working directory "
+            "as a project instead of creating a new project?."
```

Comment:
```suggestion
            "Do you want to adapt current working directory "
            "to a Protostar project instead of creating a new project?."
```

Reply:
(Maybe `convert` is a better word. `Adapt as` sounds wrong to me, but I suck at English.)

Reply:
I think adapt is better, we adjust. not transform